MapboxVisionTests/RecordCoordinatorTests.swift

```diff
@@ -72,6 +70,83 @@ final class RecordCoordinatorTests: XCTestCase {
         XCTAssert(directoryExists(at: recordingPath), "Recording should be saved at \(recordingPath) after recording is stopped")
     }
 
+    private class TestClipRequest {
+        let start: Float
+        let end: Float
+
+        init(start: Float, end: Float) {
+            self.start = start
+            self.end = end
+        }
+    }
+
+    private class RecordDelegate: RecordCoordinatorDelegate {
+        private let onRecordingStopped: () -> Void
+
+        init(onRecordingStopped: @escaping () -> Void) {
+            self.onRecordingStopped = onRecordingStopped
+        }
+
+        func recordingStarted(path: String) {
+        }
+
+        func recordingStopped(recordingPath: RecordingPath) {
+            onRecordingStopped()
+        }
+    }
+
+    private func performVideoClipping(with requests: [TestClipRequest], _ message: String = "", line: UInt = #line) {
+        let expectation = XCTestExpectation(description: "All the requested clips are successfully cut")
+        let videoSource = MockVideoSource()
+        videoSource.add(observer: coordinator)
+        videoSource.start()
+
+        let recordDelegate = RecordDelegate {
+            let path = DocumentsLocation.currentRecording.path
+            let directoryContent = try? FileManager.default.contentsOfDirectory(at: URL(string: path)!, includingPropertiesForKeys: nil, options: [])
+
+            if let directoryContent = directoryContent {
```

Comment:
`guard` may help to make code more linear. WDYT?

Reply:
I want to have one point of expectation fulfill